dockerfile_parse/parser.py

```diff
@@ -91,6 +93,10 @@ def __init__(self, path=None, cache_content=False, env_replace=True):
 
         self.env_replace = env_replace
 
+        if parent_env:
+            logger.debug("Setting inherited parent image ENV vars: %s", parent_env)
+            self.envs(parent_env)
```

Comment:
I was thinking it would be better to change line 293, where `envs = {}`.

We want to know which ENV instructions are declared in the Dockerfile, but when we also want to be able to know correct values for instructions that perform environment variable substitution such as ENV and LABEL.